tests/Unit/DataStructures/Test_Variables.cpp

```diff
@@ -547,68 +794,158 @@ SPECTRE_TEST_CASE("Unit.DataStructures.Variables.SliceVariables",
+                          make_not_null(&dist));
+
+  std::array<VectorType, 3> slice1_vals;
+  std::fill(slice1_vals.begin(), slice1_vals.end(), VectorType{2});
+  fill_with_random_values(make_not_null(&slice1_vals), make_not_null(&gen),
+                          make_not_null(&dist));
+
+  using Tensor = typename VariablesTestTags_detail::tensor<VectorType>::type;
   const Index<2> extents{{{4, 2}}};
-  Variables<tmpl::list<VariablesTestTags_detail::vector>> vars(
+  Variables<tmpl::list<VariablesTestTags_detail::tensor<VectorType>>> vars(
       extents.product());
-  get<VariablesTestTags_detail::vector>(vars) =
-      Vector{{{{1110000., 1120000., 1130000., 1140000., 1210000., 1220000.,
-                1230000., 1240000.},
-               {2110000., 2120000., 2130000., 2140000., 2210000., 2220000.,
-                2230000., 2240000.},
-               {3110000., 3120000., 3130000., 3140000., 3210000., 3220000.,
-                3230000., 3240000.}}}};
-
+  get<VariablesTestTags_detail::tensor<VectorType>>(vars) =
+      Tensor{{{orig_vals[0], orig_vals[1], orig_vals[2]}}};
   {
     const auto slice_extents = extents.slice_away(0);
-    Variables<tmpl::list<VariablesTestTags_detail::vector>> slice(
+    Variables<tmpl::list<VariablesTestTags_detail::tensor<VectorType>>> slice(
         slice_extents.product(), 0.);
-    get<VariablesTestTags_detail::vector>(slice) =
-        Vector{{{{1100., 1200.}, {2100., 2200.}, {3100., 3200.}}}};
+    get<VariablesTestTags_detail::tensor<VectorType>>(slice) =
+        Tensor{{{slice1_vals[0], slice1_vals[1], slice1_vals[2]}}};
     add_slice_to_data(make_not_null(&vars), slice, extents, 0, 2);
   }
+
   {
     const auto slice_extents = extents.slice_away(1);
-    Variables<tmpl::list<VariablesTestTags_detail::vector>> slice(
+    Variables<tmpl::list<VariablesTestTags_detail::tensor<VectorType>>> slice(
         slice_extents.product(), 0.);
-    get<VariablesTestTags_detail::vector>(slice) = Vector{
-        {{{11., 12., 13., 14.}, {21., 22., 23., 24.}, {31., 32., 33., 34.}}}};
+    get<VariablesTestTags_detail::tensor<VectorType>>(slice) =
+        Tensor{{{slice0_vals[0], slice0_vals[1], slice0_vals[2]}}};
     add_slice_to_data(make_not_null(&vars), slice, extents, 1, 1);
   }
 
-  CHECK((Vector{{{{1110000., 1120000., 1131100., 1140000., 1210011., 1220012.,
-                   1231213., 1240014.},
-                  {2110000., 2120000., 2132100., 2140000., 2210021., 2220022.,
-                   2232223., 2240024.},
-                  {3110000., 3120000., 3133100., 3140000., 3210031., 3220032.,
-                   3233233., 3240034.}}}}) ==
-        get<VariablesTestTags_detail::vector>(vars));
+  // The slice0_vals should have been added to the second half of each of the
+  // three vectors in the tensor. The slice1_vals should have been added to
+  // entries 2 and 6 in each vector.
+  // clang-format off
+  const Tensor expected = Tensor{
```

Comment:
You can remove the ` = Tensor` and just do `const Tensor expected{` if you want (I'm fine with either, just an FYI. clang-tidy might care, not sure)